src/transformers/models/bridgetower/modeling_bridgetower.py

```diff
@@ -0,0 +1,1006 @@
+        pixel_values: Optional[torch.FloatTensor] = None,
+        pixel_mask: Optional[torch.LongTensor] = None,
+        head_mask: Optional[torch.FloatTensor] = None,
+        inputs_embeds: Optional[torch.FloatTensor] = None,
+        image_embeds: Optional[torch.FloatTensor] = None,
+        image_token_type_idx: Optional[int] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+        labels: Optional[torch.LongTensor] = None,
+    ) -> Union[Tuple[torch.Tensor], BridgeTowerModelOutput]:
+        r"""
+        labels (`torch.LongTensor` of shape `(batch_size,)`, *optional*):
+            Labels are currently not supported.
+        Returns:
+
+        Examples:
+
+        ```python
+        >>> from transformers import BridgeTowerProcessor, BridgeTowerModel
+        >>> from PIL import Image
+        >>> import requests
+
+        >>> # prepare image and text
+        >>> url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+        >>> image = Image.open(requests.get(url, stream=True).raw)
+        >>> text = "hello world"
+        >>> processor = BridgeTowerProcessor.from_pretrained("BridgeTower/bridgetower-base")
+        >>> model = BridgeTowerModel.from_pretrained("BridgeTower/bridgetower-base")
+
+        >>> inputs = processor(image, text, return_tensors="pt")
+        >>> outputs = model(**inputs)
+        >>> outputs.keys()
+        odict_keys(['text_feats', 'image_feats', 'pooler_output'])
+        ```"""
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+        image_token_type_idx = image_token_type_idx if image_token_type_idx else 1
+        irtr_len = 0
+        input_shape = input_ids.size()
+        text_embeds = self.text_transformer.embeddings(input_ids=input_ids)
+
+        if attention_mask is None:
+            attention_mask = torch.ones(input_shape, dtype=torch.long, device=self.device)
+        extend_text_masks = self.text_transformer.get_extended_attention_mask(attention_mask, input_shape, self.device)
+
+        split_index = len(self.text_transformer.encoder.layer) - self.config.num_hidden_layers + 1
+        for layer in self.text_transformer.encoder.layer[:split_index]:
+            text_embeds = layer(text_embeds, extend_text_masks)[0]
+
+        image_embeds = self.vit_model.visual.forward_pre(pixel_values.type(self.vit_model.dtype))
+        for block in self.vit_model.visual.transformer.resblocks[:split_index]:
+            image_embeds = block(image_embeds)
+        image_embeds_with_ln = self.vit_model.visual.forward_post(image_embeds.type(self.vit_model.dtype))
+
+        # first layer
+        cross_modal_text = self.cross_modal_text_transform(text_embeds)
+        text_token_type_embeddings = self.token_type_embeddings(torch.zeros(1).long().to(self.device)).expand_as(
+            cross_modal_text
+        )
```

Comment:
nit: I would split into separate lines rather than trying to do it all in one. I believe `zeros_like` can be used directly here instead (I'm assuming `cross_modal_text` is also on `self.device` here. 

```suggestion
        dummy_text_input = torch.zeros_like(cross_modal_text, dtype=torch.long)
        text_token_type_embeddings = self.token_type_embeddings(dummy_text_input)
```

Reply:
The suggested changes does not provide the same results and gave me a few shape mismatch errors. I've retained the original code.

Reply:
Please split up the code in several lines. Also `self.device` should not be used as it can be wrong if the model is split on several device, it's safer to use the device of a given weight/input.